volatility3/framework/symbols/linux/utilities/modules.py

```diff
@@ -1,77 +1,121 @@
+            The first memory module in which the address fits and the symbol name for `target_address`
 
         Kernel documentation:
             "within_module" and "within_module_mem_type" functions
         """
+        kernel = context.modules[kernel_module_name]
+
+        kernel_layer = context.layers[kernel.layer_name]
+
+        if modules[0].start != modules[0].start & kernel_layer.address_mask:
+            raise ValueError(
+                "Modules list must be gathered from `run_modules_scanners` to be used in this function"
+            )
+
         matches = []
-        seen_addresses = set()
         for module in modules:
-            _, start, end = cls.mask_mods_list(context, layer_name, [module])[0]
-            if (
-                start <= target_address < end
-                and module.vol.offset not in seen_addresses
-            ):
+            if module.start <= target_address < module.end:
                 matches.append(module)
-                seen_addresses.add(module.vol.offset)
 
-        if len(matches) > 1:
-            warnings.warn(
-                f"Address {hex(target_address)} fits in modules at {[hex(module.vol.offset) for module in matches]}, indicating potential modules memory space overlap.",
-                UserWarning,
-            )
-            return matches[0]
-        elif len(matches) == 1:
-            return matches[0]
+        if len(matches) >= 1:
+            if len(matches) > 1:
+                warnings.warn(
+                    f"Address {hex(target_address)} fits in modules at {[hex(module.start) for module in matches]}, indicating potential modules memory space overlap.",
+                    UserWarning,
+                )
+
+            symbol_name = None
+
+            match = matches[0]
+
+            if match.name == constants.linux.KERNEL_NAME:
+                symbols = list(kernel.get_symbols_by_absolute_location(target_address))
 
-        return None
+                if len(symbols):
+                    symbol_name = symbols[0]
+            else:
+                module = kernel.object("module", offset=module.offset, absolute=True)
+                symbol_name = module.get_symbol_by_address(target_address)
+
+            if symbol_name:
+                symbol_name = symbol_name.split(constants.BANG)[1]
+
+            return match, symbol_name
+
+        return None, None
 
     @classmethod
+    @Deprecation.method_being_removed(
+        "This method is not needed. The correct API for mapping kernel pointers to modules is `linux_utilities_modules.Modules.run_module_scanners`"
+    )
     def mask_mods_list(
         cls,
         context: interfaces.context.ContextInterface,
-        layer_name: str,
-        mods: Iterator[interfaces.objects.ObjectInterface],
-    ) -> List[Tuple[str, int, int]]:
+        kernel_layer_name: str,
+        mods: Iterator[extensions.module],
+    ) -> List[extensions.module]:
```

Comment:
Errr, the ocde looks like it still returns tuples?  It's fine because it's deprecated but... I think the original typing information was correct, at least for the output?

Reply:
Good catch.